omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,217 @@
+		boolean manualEntryEnabled = true;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", IDENTIFIER_TYPE_UUID);
+		autoGenerationOption.add("source", SOURCE);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		assertEquals(SOURCE, PropertyUtils.getProperty(newAutoGenerationOption, "source"));
+		assertNull(PropertyUtils.getProperty(newAutoGenerationOption, "location"));
+		assertEquals(IDENTIFIER_TYPE_UUID, PropertyUtils.getProperty(newAutoGenerationOption, "identifierType"));
+	}
+	
+	@Test(expected = org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException.class)
+	public void shouldNotCreateNewAutoGenerationOptionWithNullParameters() throws Exception {
+		long originalCount = getAllCount();
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", null);
+		autoGenerationOption.add("location", null);
+		autoGenerationOption.add("source", null);
+		autoGenerationOption.add("automaticGenerationEnabled", null);
+		autoGenerationOption.add("manualEntryEnabled", null);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		assertNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount, getAllCount());
+	}
+	
+	@Test
+	public void shouldEditAnAutoGenerationOption() throws Exception {
+		boolean automaticGenerationEnabled = false;
+		boolean manualEntryEnabled = false;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("location", NEW_LOCATION);
+		autoGenerationOption.add("source", NEW_SOURCE);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		assertEquals(NEW_LOCATION,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getLocation().getUuid());
+		assertEquals(NEW_SOURCE,
+		    identifierSourceService.getAutoGenerationOption(Integer.parseInt(getUuid())).getSource().getUuid());
```

Comment:
How sure are you that these values were not already like this, even before the rest call?

Reply:
Added the assertions to check that